modules/core/arrow-instances-data/build.gradle

```diff
@@ -3,6 +3,7 @@ dependencies {
     compile project(':arrow-annotations')
     compile project(':arrow-typeclasses')
     compile project(':arrow-data')
+    compile project(':arrow-effects')
```

Comment:
This should be the other way around. Effects should have the MonadDefer instance of Kleisli.

Reply:
I'm probably missing something but, are you sure about this? The only instances I could find on `arrow-effects-instances` are the ones from `IO`, but just because it's an effects data type. Kleisli is not. That seems to be the approach until today.

The module does not contain the `MonadDefer` / `Bracket` / `Async` / `Effect` (or any other effects Type classes) instances for all the data types in Arrow.

Reply:
@JorgeCastilloPr This breaks the module system. If I include instances for `data` then I don't want to implicitly import `effects`.

What did you need `effects` for anyway?

Reply:
Nevermind that, do a new module `arrow-data-instances-effects` instead.

Reply:
Alright. Did that as a compromise solution for now, mostly to unlock this PR. We definitely need to have that maintainers call asap to decide what to do with these instances, it doesn't look great though. Here's [the commit](https://github.com/arrow-kt/arrow/pull/1043/commits/9609ea0ce4945d4916d0ee8a107c5372b776bfdb).

I like the new approach with everything super segregated, but there are still some rough edges.

Reply:
Yep! removed now, sorry. I noticed some mins ago. The commit links got broken cause of rebases, you'd need to look at complete diff though.